pyproject.toml

```diff
@@ -67,6 +67,7 @@ testpaths = ["tests"]
 norecursedirs = ["tests/non-pytest"]
 filterwarnings = [
     "error",
+    "ignore:'SearchQuery'*:DeprecationWarning",
```

Comment:
I am against us ignoring our own warnings in our testsuite. We should resolve whatever warnings we have, or appropriate tests should be updated to apply a warning filter or (even better) to test for and confirm the warning emissions with `pytest.warns`.

Reply:
@sirosen I'll go through the tests and update them as necessary. The SearchQuery class is used in lot of places. If I turn deprecations to warnings it gives an idea of the number of errors.

```
src/globus_sdk/services/search/data.py:118: 2 warnings
tests/unit/helpers/test_search.py: 11 warnings
  /Users/blackbird/Projects/00_globus/00_automate/globus-sdk-python/src/globus_sdk/services/search/data.py:118: RemovedInV4Warning: 'SearchQuery' is deprecated. Use 'SearchQueryV1' instead.
    exc.warn_deprecated("'SearchQuery' is deprecated. Use 'SearchQueryV1' instead.")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Reply:
My opinion is that that's simply the work required here, in that case.

If we take a shortcut here, we're simply making someone do that work later. Right now, the problem is obvious and apparent. We should solve it correctly today.

Reply:
btw, I'm not advocating doing the wrong thing today. Happy to work through the tests and do it right 👍. I was interested in knowing the blast radius so now that I have numbers to count down as I work I know I've hit them all 😄 .

Reply:
Sure, it's not an accusation, after all!

It's more or less what follows from my stance that we shouldn't be ignoring our own warnings.
If we don't agree that it's wrong to have the ignore rule, then it's fine to leave it in place. But if we _do_ agree that the ignore rule is wrong, then we should fix it now.

Reply:
Perhaps this could be addressed via an auto-used fixture?

Reply:
You mean a module-scoped ignore fixture for this? Possibly, if that's what's appropriate.
But if some of these are constructed in the module context (e.g., parametrize arguments), that won't work cleanly.

Reply:
@sirosen I added the `filterwarning` marker where needed and split off the SearchQuery legacy helper into a separate test where required. At this point, I should have covered all the comments.